kivy/core/video/video_ffpyplayer.py

```diff
@@ -294,65 +317,101 @@ def _next_frame_run(self):
                     frame, val = ffplayer.get_frame(force_refresh=True)
                 finally:
                     ffplayer.set_pause(bool(self._state == 'paused'))
+                    # todo: this is not safe because user could have updated
+                    # volume between us reading it and setting it
                     ffplayer.set_volume(self._volume)
             # Get next frame regular:
             else:
                 frame, val = ffplayer.get_frame()
 
             if val == 'eof':
-                sleep(0.2)
                 if not did_dispatch_eof:
                     self._do_eos()
                     did_dispatch_eof = True
+                wait_for_wakeup(None)
```

Comment:
If waiting here forever, loop eos does sometimes not work. it seems to start video again, only audio is playing and the loop after never happens.

eof also seems only be retured one time per video loop, so i changed to wait_for_wakeup(1 / 30.) and the video loops since then as it should.

Reply:
UPDATE: actually sometimes it gets val == 'eof' twice and did_dispatch_eof is True, then it runs into the wait which never gets signalled.

Reply:
I just had to reset `did_dispatch_eof` when seeked to fix this I believe. Can you check if that fixes the multiple eof issue?

Reply:
Just tested this change, it seems to work now